tools/tpm2_activatecredential.c

```diff
@@ -24,30 +25,26 @@
 
 typedef struct tpm_activatecred_ctx tpm_activatecred_ctx;
 struct tpm_activatecred_ctx {
-
-    struct {
-        UINT8 i : 1;
-        UINT8 o : 1;
-    } flags;
-
     struct {
-        char *auth_str;
-        tpm2_session *session;
-        const char *ctx_arg;
-    } key;
+        const char *ctx_path;
+        const char *auth_str;
+        tpm2_loaded_object object;
+    } keyhandle; //Typically EK
 
     struct {
-        char *auth_str;
-        tpm2_session *session;
-    } endorse;
+        const char *ctx_path;
+        const char *auth_str;
+        tpm2_loaded_object object;
+    } activatehandle; //Typically AK
```

Comment:
again more than a handle, name it based on the object it represents not the item it represents, that's what the struct members are for.

Reply:
activatehandle_obj?